changelogs/unreleased/config-anonymous-replica.md

```diff
@@ -2,9 +2,3 @@
 
 * Effectively supported `replication.anon` option, which was broken in several
   ways before (gh-9432).
-
-  There are caveats that are not resolved yet:
```

Comment:
I find it a little strange to add this list to the changelog and then remove it at the end of the PR. Since the items in the list don't look large enough to have their own changelog entry, it would be more natural to have them in the description of the PR and/or in the commit message of the first commit. Although it's just a thought, feel free to ignore it.

Reply:
I would like to make the resulting changelog a kind of 'anonymous replicas are now supported' (because it is likely what a user of 3.0.0-beta1 actually need to know). I can just add it in the last commit -- as I see we usually proceed this way -- but my idea was that a bit more verbose variant shouldn't hurt.

Reply:
This list is also present in the documentation request (last commit) and in the pull request description (formulated as constrainsts and nuances).